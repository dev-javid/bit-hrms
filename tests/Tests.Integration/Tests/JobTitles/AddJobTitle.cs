using Application.JobTitles.Commands;

namespace Tests.Integration.Tests.JobTitles
{
    public class AddJobTitle(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/job-titles";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            var command = new AddJobTitleCommand
            {
                DepartmentId = 3000,
                Name = "Test",
            };
            HttpResponseMessage response;
            using (StringContent content = new StringContent(JsonSerializer.Serialize(command), Encoding.UTF8, JsoneMediaType))
            {
                response = await Client.PostAsync(Route, content);
            }

            var result = await response.Content.ReadAsStringAsync();
            response.Dispose();
            Approvals.VerifyJson(result.Ignore("$.id"));
        }
    }
}
