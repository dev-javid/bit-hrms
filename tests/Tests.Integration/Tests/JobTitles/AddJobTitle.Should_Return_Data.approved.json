{
    "id": "__IGNORED_VALUE__"
}